### DISCUSSION TITLE
hang with motors and encoder disabled

### DISCUSSION THREAD

**Discussion by kotovasia5120** [kachurovskiy/nanoels#217] [General]
Hanging with the motors and encoder turned off is normal.
The output from the port is as follows.
`ESP-ROM:esp32s3-20210327
Build:Mar 27 2021
rst:0x1 (POWERON),boot:0x2b (SPI_FAST_FLASH_BOOT)
SPIWP:0xee
mode:DIO, clock div:1
load:0x3fce3818,len:0x109c
load:0x403c9700,len:0x4
load:0x403c9704,len:0xb50
load:0x403cc700,len:0x2fd0
entry 0x403c98ac
E (4524) pcnt(legacy): _pcnt_get_counter_value(101): PCNT driver error
E (4524) pcnt(legacy): _pcnt_get_counter_value(101): PCNT driver error`
https://github.com/kachurovskiy/nanoels/blob/main/h4/h4.ino
Added i2c display instead of standard.
Also, notifications about pressed keys appear periodically during startup.

**Comment by kachurovskiy** [comment]
Hi @kotovasia5120, 

> Hanging with the motors and encoder turned off is normal.

Not sure what this means. Please provide a more complete description of the problem, ideally with a video showing what's wrong.

> Hanging with the motors and encoder turned off is normal.

Not sure what's going on there. Nothing on Google for this error. Seems like you've modified the code significantly, try switching back to original h4.ino and checking if the problem still exists?

**Reply by kotovasia5120** [reply]
Now the board is lying on the table and only the display and keyboard are connected to it. Is it normal that the controller freezes when changing modes?

**Reply by kachurovskiy** [reply]
No, it's a sign that there's some kind of concurrency problem in the code - they usually lead to hangs.

**Reply by kotovasia5120** [reply]
also endless beeping from the speaker.

**Reply by kotovasia5120** [reply]
> also endless beeping from the speaker.

@kachurovskiy
On version espressif 2 it stopped compiling.
On the version from @0unrouted0 there is also an endless beep.
With a test sketch for beeping it works as expected.
In the code I inserted an output to the port for each beep, but there is no output, but the beeping occurs.

**Reply by kachurovskiy** [reply]
@kotovasia5120 I have no information which code you're running and on which hardware, it's not possible for me to advise you on the fact that you hear a beep.

What I need is schematics, code you're using, photos and videos of your build.
